src/main/scala/chisel3/simulator/ChiselSimSettings.scala

```diff
@@ -0,0 +1,174 @@
+package chisel3
+package simulator
+
+import svsim._
+
+/**
+  * Wrapper for unifying compilation and runtime options for different backends
+  *
+  * @param backend                      selected [[svsim.Backend]]
+  * @param verilogPreprocessorDefines   SystemVerilog defines as a map of `MACRO`->definition
+  * @param optimizationStyle
+  * @param availableParallelism
+  * @param defaultTimescale
+  * @param libraryExtensions
+  * @param libraryPaths
+  * @param disabledWarnings
+  * @param disableFatalExitOnWarnings
+  * @param traceStyle
+  * @param outputSplit
+  * @param outputSplitCFuncs
+  * @param enableAllAssertions
+  * @param customSimulationWorkingDirectory
+  * @param verboseCompile
+  * @param verboseRun
+  * @param workspaceRoot
+  * @param workingDirectoryPrefix
+  * @param executionScriptLimit
+  * @param executionScriptEnabled
+  * @param conservativeCommandResolution
+  * @param resetWorkspace                 completely delete workspace folders and their content before running the simulation
+  * @param randomlyInitializeRegisters
+  * @param chiselArgs
+  * @param firtoolArgs
+  */
+case class ChiselSimSettings[B <: Backend](
```

Comment:
I don't think we should make this a case class. They are so much more convenient to write but they make adding fields in a backwards compatible way very painful (generated unapply and copy methods are the bane of source and binary compatibility). I assume we're going to want to add settings to this quite a bit, so we should probably set ourselves up to make that easy.